### PR TITLE
Update LSP4J version to 0.15.0

### LINKED ISSUE #38108: [Improvement]: Update LS to use LSP4J 0.14
### Description

LSP4J 0.14 was released in May 2022 after the LSP 3.17 was marked as done. We should update LS to this version. It will help us introduce features like `inlay hints` that was added in 3.17.

### Related area

-> Editor

### PR DESCRIPTION
**Pull request by mindula**
## Purpose
This PR updates the LSP4J version of language server to 0.15.0

Fixes #38108

VScode plugin PR: 
* https://github.com/wso2-enterprise/ballerina-plugin-vscode/pull/661 (closed)
* https://github.com/wso2-enterprise/ballerina-plugin-vscode/pull/910

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by malinthar — CHANGES_REQUESTED** [review]
Let's link the related PR for vscode-plugin repo as well

**Review by malinthar — DISMISSED** [review]
LGTM!

**Review by IMS94 — DISMISSED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaNotebookDocumentService.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://wso2.com) All Rights Reserved.
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaNotebookDocumentService.java:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (http://wso2.com) All Rights Reserved.
```

**Review by malinthar — CHANGES_REQUESTED** [review]

**Review by malinthar — APPROVED** [review]
LGTM!

**Review by IMS94 — APPROVED** [review]

**Pull request merged**